lib/APPRemoteControl/src/App.cpp

```diff
@@ -384,3 +386,33 @@ void App_statusChannelCallback(const uint8_t* payload, const uint8_t payloadSize
         application->systemStatusCallback(currentStatus->status);
     }
 }
+
+/**
+ * Receives Turtle speed setpoints over SerialMuxProt channel.
+ *
+ * @param[in] payload       Linear and angular speed setpoints in a TurtleSpeed structure.
+ * @param[in] payloadSize   Size of the TurtleSpeed structure.
+ * @param[in] userData      Instance of App class.
+ */
+void App_turtleChannelCallback(const uint8_t* payload, const uint8_t payloadSize, void* userData)
+{
+    (void)userData;
+    if ((nullptr != payload) && (TURTLE_CHANNEL_DLC == payloadSize))
+    {
+        const TurtleSpeed* turtleSpeedData = reinterpret_cast<const TurtleSpeed*>(payload);
+        DifferentialDrive& diffDrive       = DifferentialDrive::getInstance();
+        int16_t            angularSpeed    = static_cast<int16_t>(turtleSpeedData->angular);
+
+        /* Convert to [steps/s] */
+        int16_t centerSpeed = Util::millimetersPerSecondToStepsPerSecond(turtleSpeedData->linearCenter);
+
+        /* Linear speed is set first. Overwrites all speed setpoints. */
+        diffDrive.setLinearSpeed(centerSpeed);
```

Comment:
Interfaces are mixed now ... here diffDrive is used and for SpeedData the target speed is set via DrivingState::setTargetSpeeds().